File: 0875-koko-eating-bananas/0875-koko-eating-bananas.cpp
class Solution {
public:
    int getRounding(int banana, int k) {
        if (banana % k == 0) {
            return banana / k;
        }
        return (banana / k) + 1;
    }
    
    int minEatingSpeed(vector<int>& piles, int h) {
        sort(piles.begin(), piles.end());
        int left = 1;
        int right = piles[piles.size() - 1];
        
        while (left < right) {
            int k = left + ((right - left) / 2);
            
            int sum = 0;
            for (auto &i : piles) {
                sum += getRounding(i, k);
                if (sum > h) {
                    left = k + 1;
                    continue;
                }
            }
            
            if (sum > h) {
                left = k + 1;
            }
            else {
                right = k;
            }
        }
        
        return left;
    }
};

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.kt
class Solution {
    fun getRounding(bananas: Int, k: Int): Int = 
        when (bananas % k) {
            0 -> (bananas / k)
            else -> (bananas / k) + 1
        }
    
    fun minEatingSpeed(piles: IntArray, h: Int): Int {
        piles.sort()
        var left: Int = 1
        var right: Int = piles[piles.size - 1]
        
        while (left < right) {
            val k = left + ((right - left) / 2)
            val sum = piles.map { getRounding(it, k) }.sum()
            
            when (sum > h) {
                true -> left = k + 1
                else -> right = k
            }
        }
        
        return left
    }
}